srcjs/_start.js

```diff
@@ -3,6 +3,17 @@
 
   var exports = window.Shiny = window.Shiny || {};
 
+  (function(history){
+    var pushState = history.pushState;
+    history.pushState = function(state) {
+        if (typeof history.onpushstate == "function") {
```

Comment:
Hmm, I think I don't fully follow you on this. I thought your comment meant that I should move it further into the app, so I did in commit [71a63119f05ee055369ee2f9ee3fb21e1f80d724](https://github.com/rstudio/shiny/commit/71a63119f05ee055369ee2f9ee3fb21e1f80d724) - also making sure that history.pushState has been called and the URL has changed. But we still have the same problem. Which makes me think I haven't really understood your comment.